src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -74,7 +74,7 @@ protected function processValue($value, $isRoot = false)
                 $this->unusedBindings[$bindingId] = array($key, $this->currentId);
             }
 
-            if (isset($key[0]) && '$' === $key[0]) {
+            if (preg_match('/^(?:(?:array|bool|float|int|string) )?\$/', $key)) {
```

Comment:
This needs to contain what we can put in parameters. So not iterable nor callable.

Reply:
Theoretically, a factory definition could return a callable or iterable and be used in a binding though. Do we care? (is it actually officially supported?)

Also, a class might have an iterable typehint but expected injected value be an array. Shouldn't bindings support this too?

Reply:
It's supported since it always worked yes, and it works here also, this just skips the "must be a ref" check for scalar|array types.

> a class might have an iterable typehint but expected injected value be an array

That's not supported right now. Could be added in another PR for sure (if worth it)